docs/Translation.md

```diff
@@ -12,27 +12,37 @@ The react-admin interface uses English as the default language. But it also supp
 To handle translations, the `<Admin>` component supports:
 
 - a `locale` prop expecting a string ('en', 'fr', etc), and
-- a `messages` prop, expecting a dictionary object.
+- a `messagesProvider` prop, expecting an `messagesProvider` as explained below.
+
+## `messagesProvider`
+
+Provides messages for a specific locale. React-admin ships with a `createMessagesProvider` factory to easily create your own `messagesProvider`. 
+`createMessagesProvider` requires a single argument `messageBundles` and allows an optional `defaultMessageBundle`. 
```

Comment:
I'm not fan of this `createMessagesProvider`. It obfuscates the logic behind the provider, and it introduces a second way of doing i18n - which is hard to explain. 

I think you can document how to pass custom messages simply using the following example:

```js
const messages = {
  en: englishMessages,
  fr: frenchMessages,
};
const i18nProvider = (type, params) => {
    if (type === 'GET_DEFAULT_MESSAGES') {
        return englishMessages;
    }
    if (type === 'CHANGE_LOCALE') {
        return messages[params.locale];
    }
    throw new Error('Undefined action type', type); 
}
```